### DISCUSSION TITLE
Html -> databinding based on Object key: value.

### DISCUSSION THREAD

**Discussion by khanmubeen** [richard-scryber/scryber.core#94] [Q&A] [answered]
Hi, I'm familiar with es6. In order to filter Array Object could be done with .filter(). 
let selectCars = cars.filter(car => car.type=== "station wagon");
console.log(selectCars);
// output from below Object would return with type: "station wagon":
// [
//   {
//     color: 'red',
//     type: 'station wagon',
//     registration: 'Sat Mar 03 2018 01:00:00 GMT+0100 (GMT+01:00)',
//     capacity: 5
//   },
//   {
//     color: 'red',
//     type: 'cabrio',
//     registration: 'Sat Mar 03 2012 01:00:00 GMT+0100 (GMT+01:00)',
//     capacity: 2
//   }
// ]

Question: How could I achieve this on Html so single 
html table contains data from type:"station wagon". Another table from given type.
Backend sends a json object, which I'm taking into use -> https://scrybercore.readthedocs.io/en/latest/7_extending/10_binding_parameters.html#combining-selector-paths.
Thank you for your valuable input and time into this matter.
/Mubeen

**Comment by richard-scryber** [comment, marked as answer]
Hi Mubeen

In the latest beta releases of the package we do support a few coalescing and aggregate functions.
Each, eachOf, selectWhere, and sortBy

The selectWhere sounds like it would work for your situation e.g.

<template data-bind="{{selectWhere(model.items, .type =='station wagon')}}" ></template>

They are not well documented, but there is a unit test sample if you wanted to check it in action, Last unit test from here

[Expression Binding Test](https://github.com/richard-scryber/scryber.core/blob/master/Scryber.UnitTest/Binding/ExpressionBinding_Test.cs)



And I have added a sample to our new preview tool, which will hopefully work for you - would love to know if it does.

[Station wagon preview](https://www.paperworkday.net/preview.html?ui=Paging%2CZoom%2CCode%2CEdit&template=https%3A%2F%2Fraw.githubusercontent.com%2Frichard-scryber%2FPaperworkDayDocs%2Fmain%2Fdocs%2F_samples%2Fselectbinding%2Fselecttemplate.html&data=https%3A%2F%2Fraw.githubusercontent.com%2Frichard-scryber%2FPaperworkDayDocs%2Fmain%2Fdocs%2F_samples%2Fselectbinding%2Fselectdata.json)

You can see / alter the template and data by clicking the edit button.

**Reply by khanmubeen** [reply]
Thank you Richard. Template example is nicely done.

**Reply by richard-scryber** [reply]
Glad that it worked for you Mubeen, Richard :-)
